indexd/drs/blueprint.py

```diff
@@ -247,19 +247,20 @@ def bundle_to_drs(record, expand=False, is_content=False):
 
 
 def parse_checksums(record, drs_object):
-    if "hashes" in record:
-        for k in record["hashes"]:
-            drs_object["checksums"].append({"checksum": record["hashes"][k], "type": k})
-    else:
-        if "checksums" in record:
-            for checksum in record["checksums"]:
-                drs_object["checksums"].append(
+    ret_checksum = []
+    try:
+        if "hashes" in record:
+            for k in record["hashes"]:
+                ret_checksum.append({"checksum": record["hashes"][k], "type": k})
+        elif "checksum" in record:
+            checksums = json.loads(record["checksum"])
+            for checksum in checksums:
+                ret_checksum.append(
                     {"checksum": checksum["checksum"], "type": checksum["type"]}
                 )
-        else:
-            drs_object["checksums"].append(
-                {"checksum": record["checksum"], "type": "md5"}
-            )
+        return ret_checksum
+    except KeyError:
+        raise UserError("Object does not contain checksum")
```

Comment:
remove try/except/UserError => `parse_checksums` is only used on db objects so any error would be an internal error, not user error